### PR TITLE
patched bugsnag changes applied to llvm in single commit

### PR DESCRIPTION
**Pull request by DavidPetran**
Created this PR to view differences between llvm/llvm-project and forked bugsnag llvm/project. 
Delete the djp/changed-view branch instead of merging

commented the parts I've changed in this update

### CODE REVIEW

File: llvm/lib/ProfileData/RawMemProfReader.cpp [hunk 1]
```diff
@@ -487,7 +487,7 @@ Error RawMemProfReader::symbolizeAndFilterStackFrames() {
       DIInliningInfo DI = DIOr.get();
```
Comment (DavidPetran) [inline llvm/lib/ProfileData/RawMemProfReader.cpp:487]: New changes from llvm requiring locally changed DILineInfo

File: llvm/lib/DebugInfo/Symbolize/MarkupFilter.cpp [hunk 2]
```diff
@@ -278,7 +278,7 @@ bool MarkupFilter::tryPC(const MarkupNode &Node) {
   }
```
Comment (DavidPetran) [inline llvm/lib/DebugInfo/Symbolize/MarkupFilter.cpp:279]: New changes from llvm requiring locally changed DILineInfo

File: llvm/lib/DebugInfo/Symbolize/DIPrinter.cpp [hunk 3]
```diff
@@ -313,18 +329,21 @@ void JSONPrinter::print(const Request &Request, const DIInliningInfo &Info) {
   json::Array Array;
   for (uint32_t I = 0, N = Info.getNumberOfFrames(); I < N; ++I) {
     const DILineInfo &LineInfo = Info.getFrame(I);
-    json::Object Object = toJSON(LineInfo);
+    json::Object Object = toJSON(LineInfo, I > 0);
     SourceCode SourceCode(LineInfo.FileName, LineInfo.Line,
```
Comment (DavidPetran) [inline llvm/lib/DebugInfo/Symbolize/DIPrinter.cpp:333]: In llvm toJson was extracted, made the same change to reduce future merge conflicts

**Review by mikeewhite — APPROVED** [review]
LGTM

**Pull request closed without merging**